pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -663,11 +665,11 @@ func (service *SecurityPolicyService) buildRuleDisplayName(rule *v1alpha1.Securi
 		return util.GenerateTruncName(common.MaxNameLength, ruleName, "", "", "", ""), nil
 	} else {
 		// For the security policy rule with namedPort, it will be expanded to the multiple security policy rules based on resolution of named port.
-		// e.g. input: security policy's rule name: TCP.http-UDP.1234-ingress-allow,
-		// expand to NSX security policy rules with name TCP.http-UDP.1234.TCP.80-ingress-allow and TCP.http-UDP.1234.UDP.1234-ingress-allow.
-		// in case that user defined input security policy's rule name: sp-namedport-rule,
-		// expand to NSX security policy rules with name sp-namedport-rule.TCP.80-ingress-allow and sp-namedport-rule.UDP.1234-ingress-allow.
-		index := strings.Index(ruleName, "-"+suffix)
+		// e.g. input: security policy's rule name: TCP.http_UDP.1234_ingress-allow,
+		// expand to NSX security policy rules with name TCP.http_UDP.1234_TCP.80_ingress-allow and TCP.http_UDP.1234_UDP.1234_ingress-allow.
+		// in case that user defined input security policy's rule name: sp_namedport_rule,
+		// expand to NSX security policy rules with name sp_namedport_rule.TCP.80_ingress-allow and sp_namedport_rule.UDP.1234_ingress-allow.
```

Comment:
Why do we still use connector "-" between `TCP.80_ingress` an `allow` ?

Reply:
Yes, that's right, it should be `TCP.80_ingress_allow`